VideoLocker/src/main/java/org/edx/mobile/base/BaseSingleFragmentActivity.java

```diff
@@ -171,4 +166,10 @@ protected void hideMessageInSitu(){
         }
     }
 
+    private void showErrorDialog(String message) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage(message)
+                .setPositiveButton(R.string.label_ok, null)
+                .create().show();
```

Comment:
Testing these scenarios, none of them seem to produce the effect of the dialog not being dismissed upon Activity destruction, though in some cases the dialog is destroyed prematurely. I suppose the `DialogFragment` would produce more robust binding against these cases.

> Pressing home while low on memory

That would probably cause the whole process to be killed, and the dialog should be destroyed along with it. I simulated this by pressing home and then killing the app process through ADB, and the dialog was destroyed when I resumes the app.

> Switching to another app

Switching to another app doesn't destroy the Activity, but it does hide the dialog while the Activity is stopped.

> Opening an edX push notification (though, we don't have those yet...)

I simulated that by scheduling another Activity to be started on top of that dialog, and the dialog does not show on the new Activity (but it _is_ still retained when I go back to the previous Activity).

> User opens task-switcher and swipes dismisses the edX task

This kills the dialog too.

> Device enters Doze mode

I wasn't familiar with Doze mode before, and running the `adb shell dumpsys battery unplug` does seem to dismiss the dialog while retaining the Activity.